frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -168,60 +170,61 @@ function FileManagerMenu:setUpdateItemTable()
                 text = _("Classic mode settings"),
                 sub_item_table = {
                     {
-                        text = _("Items per page"),
+                        text_func = function()
+                            return T(_("Items per page: %1"), FileChooser.perpage)
+                        end,
                         help_text = _([[This sets the number of items per page in:
 - File browser, history and favorites in 'classic' display mode
 - Search results and folder shortcuts
 - File and folder selection
 - Calibre and OPDS browsers/search results]]),
-                        callback = function()
-                            local SpinWidget = require("ui/widget/spinwidget")
-                            local Menu = require("ui/widget/menu")
-                            local default_perpage = Menu.items_per_page_default
-                            local curr_perpage = G_reader_settings:readSetting("items_per_page") or default_perpage
-                            local items = SpinWidget:new{
-                                value = curr_perpage,
-                                value_min = 6,
-                                value_max = 24,
-                                default_value = default_perpage,
+                        callback = function(touchmenu_instance)
+                            local current_value = FileChooser.perpage
+                            local default_value = FileChooser.items_per_page_default
```

Comment:
(Wondered if you forgot to add changes to filechooser - but I see `items_per_page_default` is defined in Menu, which FileChooser is a subclass, so it will be found. Just saying just in case.)